### DISCUSSION TITLE
Permission issue

### DISCUSSION THREAD

**Discussion by v3rtho** [silverhack/monkey365#19] [Q&A]
Hello,

I'm testing the module on my demo tenant. My user got the follow permissions:

Exchange admin
SharePoint admin
Global reader
Security Admin

I still receive the error:

"Warninig: Unable to get token for sharepoint online"
I get this warning from the Get-TenantInfo.ps1:61 char:12.

Did you saw this behavior before?

**Comment by silverhack** [comment]
Hi Thomas, 

It seems that your SharePoint Online is not correctly configured or your user does not have enough permissions to access sites/Lists, etc.. Secondly, your user **has too many privileges** for Monkey 365. Please, see docs regarding privileges:

- [https://silverhack.github.io/monkey365/permissions/](url)

- [https://support.microsoft.com/en-us/office/overview-site-governance-permission-and-sharing-for-site-owners-95e83c3d-e1b0-4aae-9d08-e94dcaa4942e](url)

Hope it helps, 
Cheers

**Comment by v3rtho** [comment]


**Reply by silverhack** [reply]
Those are not errors. Those are warnings that is indicating that the tool did not find specific configurations in your environment.

**Comment by silverhack** [comment]
I would recommend to look at Microsoft docs for configurations regarding Exchange Online. For example, and according to the warnings, it seems that safe attachment is not configured in your tenant. 

[https://docs.microsoft.com/en-us/microsoft-365/security/office-365-security/set-up-safe-attachments-policies?view=o365-worldwide](https://docs.microsoft.com/en-us/microsoft-365/security/office-365-security/set-up-safe-attachments-policies?view=o365-worldwide)
